### PR TITLE
Enable OMR string tests

### PR DESCRIPTION
**Pull request by youngar**
- re-enable OMR string port library tests
- fix tests on OSX

### CODE REVIEW

**Comment by alin00** [comment]
Is there an issue associated with this pull request?  If not please create one.  The issue # should be referenced by the commit comments.

File: fvtest/porttest/omrstrTest.cpp [hunk 1]
```diff
@@ -1130,13 +1131,10 @@ TEST(PortStrTest, str_convU8ToPlat_Null)
 /**
  * Verify string wide->MUTF8 conversion, source string has no byte order mark.
  */
-TEST(PortStrTest, str_convWideToU8Bom)
+TEST(PortStrTest, DISABLED_str_convWideToU8Bom)
```
Comment (alin00) [inline fvtest/porttest/omrstrTest.cpp]: Why do these tests need to be disabled?

I believe they used to pass before this functionality was refactored into OMR. Has a regression been introduced at some point?

**Comment by youngar** [comment]
The two tests I disabled are unimplemented, and fail when they print this diagnostic message: `outputErrorMessage(PORTTEST_ERROR_ARGS, "test not implemented");`

**Comment by alin00** [comment]
Ok, I see that.

I have some nitpicks about the code formatting.

There appear to be a few unrelated issues grouped together into this pull request. In the future, please submit unrelated fixes separately.

File: fvtest/porttest/omrstrTest.cpp [hunk 2]
```diff
@@ -678,7 +680,7 @@ TEST(PortStrTest, str_test4)
 			sizeOfFullFormatString = sizeOfFullFormatString + 1 /* null terminator */;
 
 			/* ok, now build the format string */
-			fullFormatString = omrmem_allocate_memory(sizeOfFullFormatString, OMRMEM_CATEGORY_PORT_LIBRARY);
+			fullFormatString = (char *) omrmem_allocate_memory(sizeOfFullFormatString, OMRMEM_CATEGORY_PORT_LIBRARY);
```
Comment (alin00) [inline fvtest/porttest/omrstrTest.cpp]: extra space after (char *)

File: fvtest/porttest/omrstrTest.cpp (hunk 1 above)
Comment (alin00) [inline fvtest/porttest/omrstrTest.cpp]: Please rename this test case to DISABLED_str_convWideToU8**No**Bom. The missing "No" is a typo.

File: fvtest/porttest/omrstrTest.cpp [hunk 3]
```diff
@@ -1279,11 +1276,11 @@ TEST(PortStrTest, str_convRoundTrip)
 	uint32_t unicodeEnd = 0x10000; /* overestimate */
 	uint32_t numCodePoints = 0;
 	size_t bufferSize = (6 * unicodeEnd) + 2; /* loose upper bound */
-	U_16 *unicodeData = omrmem_allocate_memory(unicodeEnd * 2, OMRMEM_CATEGORY_PORT_LIBRARY);
-	uint8_t *mutf8Buffer = omrmem_allocate_memory(bufferSize, OMRMEM_CATEGORY_PORT_LIBRARY);
-	uint8_t *unicodeResult = omrmem_allocate_memory(bufferSize, OMRMEM_CATEGORY_PORT_LIBRARY);
+	uint16_t *unicodeData   = (uint16_t *) omrmem_allocate_memory(unicodeEnd * 2, OMRMEM_CATEGORY_PORT_LIBRARY);
+	uint8_t  *mutf8Buffer   = (uint8_t *)  omrmem_allocate_memory(bufferSize, OMRMEM_CATEGORY_PORT_LIBRARY);
+	uint8_t  *unicodeResult = (uint8_t *)  omrmem_allocate_memory(bufferSize, OMRMEM_CATEGORY_PORT_LIBRARY);
```
Comment (alin00) [inline fvtest/porttest/omrstrTest.cpp]: Please don't align the variable names, ='s, and omrmem_allocate_memory()s.  Remove the padding spaces.

File: fvtest/porttest/omrstrTest.cpp [hunk 4]
```diff
@@ -1299,19 +1296,19 @@ TEST(PortStrTest, str_convRoundTrip)
 	omrtty_printf("Testing %d code points\n", numCodePoints);
 
 	convertedStringLength = omrstr_convert(J9STR_CODE_WIDE, J9STR_CODE_MUTF8,
-										   (uint8_t *)unicodeData, 2 * numCodePoints, mutf8Buffer, bufferSize);
+										   (char *) unicodeData, 2 * numCodePoints, (char *) mutf8Buffer, bufferSize);
```
Comment (alin00) [inline fvtest/porttest/omrstrTest.cpp]: Please remove spaces after cast operators, here and elsewhere.

File: port/common/omrstr.c [hunk 5]
```diff
@@ -2594,8 +2594,16 @@ convertMutf8ToPlatform(struct OMRPortLibrary *portLibrary, const uint8_t *inBuff
...
+		uintptr_t wideBufferCount = 0;
+		int32_t platformPartCount = 0;
+		
+		int32_t result = convertMutf8ToWide(&mutf8Cursor, &mutf8Remaining, wideBuffer, wideBufferSize);
+		if (result < 0) { /* conversion error */
+			return result;
+		}
+		wideBufferCount = (uintptr_t) result;
+
+		platformPartCount = convertWideToPlatform(portLibrary,  encodingState, &wideBufferCursor, &wideBufferCount, outCursor, outLimit);
```
Comment (alin00) [inline port/common/omrstr.c]: extra space after ```portLibrary,```

**Comment by alin00** [comment]
Thank you.

**Pull request merged**